src/melee.cpp

```diff
@@ -147,49 +147,19 @@ bool player::handle_melee_wear( item &shield, float wear_multiplier )
 
 int player::get_hit_weapon( const item &weap ) const
 {
-    // TODO: there is a listing of the same skills in player.cpp
-    int unarmed_skill = get_skill_level( skill_unarmed );
-    int bashing_skill = get_skill_level( skill_bashing );
-    int cutting_skill = get_skill_level( skill_cutting );
-    int stabbing_skill = get_skill_level( skill_stabbing );
-    int melee_skill = get_skill_level( skill_melee );
-
-    if( has_active_bionic( "bio_cqb" ) ) {
-        unarmed_skill = 5;
-        bashing_skill = 5;
-        cutting_skill = 5;
-        stabbing_skill = 5;
-        melee_skill = 5;
-    }
-
-    float best_bonus = 0.0;
-
-    // Are we unarmed?
-    if( weap.has_flag("UNARMED_WEAPON") ) {
-        ///\EFFECT_UNARMED provides weapon bonus for UNARMED_WEAPON
-        best_bonus = unarmed_skill / 2.0f;
-    }
+    ///\EFFECT_UNARMED improves hit chance for unarmed weapons
+    ///\EFFECT_BASHING improves hit chance for bashing weapons
+    ///\EFFECT_CUTTING improves hit chance for cutting weapons
+    ///\EFFECT_STABBING improves hit chance for piercing weapons
+    auto bonus = get_skill_level( weap.weap_skill() );
 
-    // Using a bashing weapon?
-    if( weap.is_bashing_weapon() ) {
-        ///\EFFECT_BASHING provides weapon bonus for bashing weapons
-        best_bonus = std::max( best_bonus, bashing_skill / 3.0f );
+    // CQB bionic acts as a lower bound providing item uses a weapon skill
+    if( bonus < 5 && has_active_bionic( "bio_cqb" ) ) {
+        bonus = 5;
     }
 
-    // Using a cutting weapon?
-    if( weap.is_cutting_weapon() ) {
-        ///\EFFECT_CUTTING provides weapon bonus for cutting weapons
-        best_bonus = std::max( best_bonus, cutting_skill / 3.0f );
-    }
-
-    // Using a spear?
-    if( weap.has_flag("SPEAR") || weap.has_flag("STAB") ) {
-        ///\EFFECT_STABBING provides weapon bonus for SPEAR or STAB weapons
-        best_bonus = std::max( best_bonus, stabbing_skill / 3.0f );
-    }
-
-    ///\EFFECT_MELEE adds to other weapon bonuses
-    return int(melee_skill / 2.0f + best_bonus);
+    ///\EFFECT_MELEE improves hit chance for all items (including non-weapons)
+    return bonus + ( get_skill_level( skill_melee ) / 2.0f );
```

Comment:
What is the plan with #18351 as I don't want to hold either PR up given I've reviewed #18351 and it's otherwise ready to merge and we are also a long way along with this.

If you have an update ready for test tolerances and standard survivors then I'll wait on that, otherwise one of us can do that in a followup PR.